components/director/pkg/scope/directive.go

```diff
@@ -44,6 +49,9 @@ func (d *directive) matches(actual []string, required []string) bool {
 	actMap := make(map[string]interface{})
 
 	for _, a := range actual {
+		for _, authn := range d.authenticators {
+			a = strings.TrimPrefix(a, authn.ScopePrefix)
```

Comment:
Can it happen that there are multiple authenticators with overlapping ScopePrefix? In that case it might be better to build the map keys for each scope and each authenticator

Reply:
Sure, I missed that logic in there. I agree that it is better this way.